core/eolearn/core/eodata_io.py

```diff
@@ -95,6 +120,17 @@ def save_eopatch(
     else:
         _check_letter_case_collisions(eopatch_features, [])
 
+
+def _remove_old_eopatch(filesystem: FS, patch_location: str) -> None:
+    filesystem.removetree(patch_location)
```

Comment:
what happens if someone does `EOPatch().save('/', overwrite_permission=OverwritePermission.OVERWRITE_PATCH)` 😬

Reply:
their problem :) any other idea on how to do patch overwriting? I guess we could try and identify all possible files (since we know the expected patch structure) and remove those... But that is a lot of extra work.

Reply:
One way that comes to mind is to do a listdir and filter it with a regex pattern, which could somehow be auto-constructed from feature types. I agree it's some extra work, but doesn't seem impossible. If you think it's worth spending a couple of hours to increase the safety here, I can tackle this.

Reply:
you can open up an issue about it, and we'll see if we ever get around to doing it